src/java/main/org/apache/zookeeper/server/auth/ProviderRegistry.java

```diff
@@ -61,17 +64,17 @@ public static void initialize() {
         }
     }
 
-    public static AuthenticationProvider getProvider(String scheme) {
+    public static ServerAuthenticationProvider getProvider(ZooKeeperServer zks, String scheme) {
```

Comment:
What if we make the signature `Object obj, String scheme` or maybe a variable number of arguments `Object..., String scheme` to enable future changes to pass other instances through the `getProvider` method?